openvdb_houdini/openvdb_houdini/SOP_OpenVDB_Extrapolate.cc

```diff
@@ -0,0 +1,768 @@
+            "(sparsely represented regions of constant value) will in that case\n"
+            "be densified, which could significantly increase memory usage.\n\n"
+            "Proper signed distance fields don't have active tiles.\n"));
+
+    parms.add(hutil::ParmFactory(PRM_INT_J, "dilate", "Dilation")
+        .setDefault(3)
+        .setRange(PRM_RANGE_RESTRICTED, 0, PRM_RANGE_UI, 10)
+        .setTooltip("The number of voxels by which to dilate the level set narrow band"));
+
+    // Dilation Pattern
+    parms.add(hutil::ParmFactory(PRM_STRING, "pattern", "Dilation Pattern")
+         .setChoiceListItems(PRM_CHOICELIST_SINGLE, {
+            "NN6",  "Faces",
+            "NN18", "Faces and Edges",
+            "NN26", "Faces, Edges and Vertices"
+         })
+         .setDefault("NN6")
+         .setTooltip("Select the neighborhood pattern for the dilation operation.")
+         .setDocumentation(
+             "The neighborhood pattern for the dilation operation\n\n"
+             "__Faces__ is fastest.  __Faces, Edges and Vertices__ is slowest\n"
+             "but can produce the best results for large dilations.\n"
+             "__Faces and Edges__ is intermediate in speed and quality.\n"));
+
+    parms.add(hutil::ParmFactory(PRM_FLT_J, "isovalue", "Isovalue")
+         .setDefault(0.5)
+         .setRange(PRM_RANGE_UI, -3, PRM_RANGE_UI, 3)
+         .setTooltip("Isovalue for which the SDF is computed"));
+
+    parms.add(hutil::ParmFactory(PRM_INT_J, "sweeps", "Iterations")
+         .setDefault(1)
+         .setRange(PRM_RANGE_RESTRICTED, 1, PRM_RANGE_UI, 5)
+         .setTooltip(
+            "The desired number of iterations of the Fast Sweeping algorithm"
+            " (one is often enough)"));
+
+    // Extension fields
+    parms.add(hutil::ParmFactory(PRM_HEADING, "extensionFields", "Extension Fields")
+         .setDocumentation(
+             "These supply the fields to be extended."));
+
+    // Dynamic grid menu
+    hutil::ParmList gridParms;
+    {
+        {   // Element type menu
+            std::vector<std::string> items;
+            for (int i = 0; i < NUM_DATA_TYPES; ++i) {
+                items.push_back(dataTypeToString(DataType(i))); // token
+                items.push_back(dataTypeToMenuItems(DataType(i))); // label
+            }
+            gridParms.add(hutil::ParmFactory(PRM_STRING, "elementType#", "Type")
+                .setChoiceListItems(PRM_CHOICELIST_SINGLE, items)
+                .setTooltip("The type of value stored at each voxel")
+                .setDocumentation(
+                    "The type of value stored at each voxel\n\n"
+                    "VDB volumes are able to store vector values, unlike Houdini volumes,\n"
+                    "which require one scalar volume for each vector component."));
+        }
+
+        // Optional grid name string
+        gridParms.add(hutil::ParmFactory(PRM_STRING, "gridName#", "Name")
```

Comment:
This isn't a name, but a group.
matchgroup#, "Match Group"
It should also show up before Type because the match group will filter out which things this multparm will generate from?